app/uk/gov/hmrc/tai/connectors/DesConnector.scala

```diff
@@ -50,44 +49,49 @@ class DesConnector @Inject()(
 
   def daPtaOriginatorId = config.daPtaOriginatorId
 
-  def desPathUrl(nino: Nino, path: String) = s"${config.baseURL}/pay-as-you-earn/individuals/$nino/$path"
+  private def desPathUrl(nino: Nino, path: String) = s"${config.baseURL}/pay-as-you-earn/individuals/$nino/$path"
 
-  def commonHeaderValues =
+  private def createDesHeader(implicit hc: HeaderCarrier) =
     Seq(
-      "Environment"   -> config.environment,
-      "Authorization" -> config.authorization,
-      "Content-Type"  -> TaiConstants.contentType)
-
-  def header: HeaderCarrier = HeaderCarrier(extraHeaders = commonHeaderValues)
-
-  def headerForUpdate(version: Int, originatorId: String): HeaderCarrier =
-    HeaderCarrier(extraHeaders = commonHeaderValues ++ Seq("Originator-Id" -> originatorId, "ETag" -> version.toString))
+      "Environment"          -> config.environment,
+      "Authorization"        -> config.authorization,
+      "Content-Type"         -> TaiConstants.contentType,
+      HeaderNames.xSessionId -> hc.sessionId.fold("-")(_.value),
+      HeaderNames.xRequestId -> hc.requestId.fold("-")(_.value),
+      "CorrelationId"        -> UUID.randomUUID().toString
+    )
+
+  private def header(implicit hc: HeaderCarrier): HeaderCarrier = HeaderCarrier(extraHeaders = createDesHeader)
+
+  private def headerForUpdate(version: Int, originatorId: String)(implicit hc: HeaderCarrier): HeaderCarrier =
+    HeaderCarrier(
+      extraHeaders = createDesHeader ++ Seq(
+        "Originator-Id" -> originatorId,
+        "ETag"          -> version.toString
+      ))
 
   def getIabdsForTypeFromDes(nino: Nino, year: Int, iabdType: Int)(
     implicit hc: HeaderCarrier): Future[List[NpsIabdRoot]] = {
     val urlToRead = desPathUrl(nino, s"iabds/tax-year/$year?type=$iabdType")
-    implicit val hc: HeaderCarrier = header
-    getFromDes[List[NpsIabdRoot]](urlToRead, APITypes.DesIabdSpecificAPI).map(x => x._1)
+    getFromDes[List[NpsIabdRoot]](urlToRead, APITypes.DesIabdSpecificAPI)(hc = header, implicitly).map(x => x._1)
```

Comment:
This would be cleaner if we changed getFromDes to take in an explicit `headers: Seq[String, String]` parameter and left the original hc in tact

Reply:
Done